python/lancedb/rerankers/cohere.py

```diff
@@ -0,0 +1,51 @@
+from typing import Union
+from functools import cached_property
+from .base import Reranker
+import lancedb
+import pyarrow as pa
+import os
+
+
+class CohereReranker(Reranker):
+    """
+    Reranks the results using cohere rerank api.
+
+    Parameters
+    ----------
+    model_name : str, default "rerank-multilingual-v2.0"
+        The name of the cross encoder model to use. Available cohere models are:
+        - rerank-english-v2.0
+        - rerank-multilingual-v2.0
+    column : str, default "text"
+        The name of the column to use as input to the cross encoder model.
+    top_k : str, default None
+        The number of results to return. If None, will return all results.
+    """
+    def __init__(self, model_name:str = "rerank-multilingual-v2.0", column:str = "text", top_n: Union[int, None]=None):
+        self.model_name = model_name
+        self.column = column
+        self.top_n = top_n
+        
+    @cached_property
+    def client(self):
+        cohere = self.safe_import("cohere")
+        if os.environ.get("COHERE_API_KEY") is None:
+                self.api_key_not_found_help("cohere")
+        return cohere.Client(os.environ["COHERE_API_KEY"])
+    
+    def rerank_hybrid(self, query_builder: "lancedb.HybridQueryBuilder" , vector_results: pa.Table, fts_results: pa.Table, combined_results: pa.Table):
+        docs = combined_results[self.column].to_pylist()
+        results = self.client.rerank(query=query_builder._query, documents=docs, top_n=self.top_n, model=self.model_name)
+        results = [(result.index, result.relevance_score) for result in results] 
+        # sort by score
+        results = sorted(results, key=lambda x: x[1], reverse=True)
```

Comment:
have you timed the python sort by key? IIRC it's pretty slow.
should be faster to create two numpy arrays one for index and the other for relevance score then use np.argsort.

Reply:
haven't benchmarked anything yet

Reply:
I agree, np.argsort is just so much better for this task than Python's sorted